### ISSUE TITLE
Decrease the default deadzone value for actions to better match real world use cases

### ISSUE THREAD

**Issue by stephanbogner** [godotengine/godot-proposals#7069]
### Describe the project you are working on

A game allowing a gamepad as input.

### Describe the problem or limitation you are having in your project

The default deadzone value is `0.5`. By being the default value this implies that it's a reasonable choice.

[Other people recommend a much lower deadzone value (0.05)](https://github.com/godotengine/godot-proposals/issues/3709) and even the [Godot docs use 0.2 as an example value](https://docs.godotengine.org/en/stable/classes/class_inputmap.html#class-inputmap-method-add-action).
According to my experience this also behaves weirdly (on my Xbox 360 controller this feels more like using a d-pad than a stick). This might also confuse newer devs that have never heard about deadzones.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Change the default value to something that is more realistic but also works with older/worse hardware.
Maybe to the `0.2` mentioned in the Godot docs.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Change the default value.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Basically everyone must change the setting in the input map.

### Is there a reason why this should be core and not an add-on in the asset library?

It's already core, it's just about changing the default.

**Comment by and-rad** [comment]
Sounds reasonable to me. If I remember correctly, UE uses something like 0.25 and Unity somewhere around 0.15.

**Comment by Calinou** [comment]
One caveat with decreasing the deadzone by a lot is that it also affects "binary" actions associated to axes. Do we want the deadzone to be only 0.2 for those binary actions, e.g. when you associate the right trigger to a Shoot action?

**Comment by Calinou** [comment]
> Good point regarding binary actions! Would it technically be possible to separate them?

This would add a lot of complexity to the deadzone implementation (what if you bind both a stick and a trigger to a single action), so I'd prefer not.

**Comment by jancborchardt** [comment]
Did a bit of trigger sensitivity testing with the Godot joypads tool and an Xbox Series controller.
These are the values at which the triggers activate the action, sorted by sensitivity. Sensitivity was always the same for both triggers:
- Dead Cells: **0.363**
- Hades: **0.268**
- Hollow Knight: **0.213**
- CS:GO: **0.174**
- Hunt:Showdown: **0.120**

Feel free to add to the list so we can figure out more. :) From the sample it seems that indeed something around **0.2** could be a better default. Let me know if I should do more or different testing.

**Comment by stephanbogner** [comment]
Thanks for the investigation @jancborchardt!

Honestly I don't have enough experience to conclude anything. I think everything lower than the current `0.5` is an improvement. So: `0.2` sounds great to me, but `0.25` would also be fine.

@Calinou:
Do you want to decide this? Or how should we move forward?
And if we move forward should I create a PR with the chosen default value?

**Comment by Calinou** [comment]
> Do you want to decide this? Or how should we move forward?
And if we move forward should I create a PR with the chosen default value?

Feel free to open a PR with `0.2` as the default value :slightly_smiling_face: 

But as I said, it must *not* interfere with existing projects in any way as I said above:

> Only built-in actions and newly created actions should be affected by this change.

**Comment by stephanbogner** [comment]
@Calinou Sounds great.
I am gonna dig into the code base (haven't contributed yet, but looking forward) :slightly_smiling_face:

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/97281.

**Comment by h0lley** [comment]
I see this has been partially reverted. not sure where discussion would best continue but this thread has the most participants so I guess it makes sense here.

anyway, I think 0.2 is much too low not just for toggle actions, many sticks trigger that in neutral position after they have went through some usage. I think 0.25 is a minimum, but something between 0.3 ~ 0.4 is probably more reasonable.
Dead Cells's 0.363 seems good to me. Not sure why it's this exact number, perhaps 0.35 is the way to go. then perhaps no DEFAULT_TOGGLE_DEADZONE would be needed.
